File: Cpp/101.cpp
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        if(!root){
            return true;
        }
        
        return isSameNode(root->left, root->right);
    }
    
    bool isSameNode(TreeNode* left, TreeNode* right){
        if(!left && !right){
            return true;
        }
        if(!left || !right){
            return false;
        }
        if(left->val != right->val){
            return false;
        }
        
        return isSameNode(left->left, right->right) &&
            isSameNode(left->right, right->left);
    }
};

File: Python/101.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def isSymmetric(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if root == None:
            return True
        return self.isMirror(root.left, root.right)

    def isMirror(self, p, q):
        if p == None and q == None:
            return True
        if p == None:
            return False
        if q == None:
            return False
        return p.val == q.val and self.isMirror(p.left, q.right) and self.isMirror(p.right, q.left)
